dso-l2/src/main/java/com/tc/objectserver/entity/EntityManagerImpl.java

```diff
@@ -189,21 +195,33 @@ public boolean removeDestroyed(EntityID id) {
       snapshotLock.release();
     }
   }
-
+  
   @Override
-  public Optional<ManagedEntity> getEntity(EntityID id, long version) throws EntityException {
+  public Optional<ManagedEntity> getEntity(EntityDescriptor descriptor) throws EntityException {
+    if (descriptor.isIndexed()) {
+      return getEntity(descriptor.getFetchID());
+    } else {
+      return getEntity(descriptor.getEntityID(), descriptor.getClientSideVersion());
+    }
+  }
+  
+  private Optional<ManagedEntity> getEntity(long idx) {
```

Comment:
Can these names be made more descriptive?  They are currently both called "getEntity" but interpret their arguments very differently.